client/client.go

```diff
@@ -0,0 +1,85 @@
+package client
+
+import (
+	"crypto/tls"
+	"errors"
+	"log/slog"
+	"net/url"
+
+	"github.com/go-ldap/ldap/v3"
+	"github.com/lucat1/me/config"
+)
+
+func Start() (*ldap.Conn, error) {
+	conf := config.Get()
+
+	ldapUrl, err := url.Parse(conf.LdapConfig.Address)
+	if err != nil {
+		return nil, err
+	}
+
+	tlsConfig := tls.Config{InsecureSkipVerify: conf.LdapConfig.InsecureSkipVerify}
+
+	var c *ldap.Conn = nil
+
+	switch ldapUrl.Scheme {
+	case "ldap":
+		{
+			slog.With("scheme", ldapUrl.Scheme)
+			if conf.LdapConfig.StartTLS {
+				slog.With("address", conf.LdapConfig.Address).
+					Debug("Trying to connet to ldap server...")
+				c, err = ldap.DialURL(conf.LdapConfig.Address)
+				if err != nil {
+					return nil, err
+				}
+
+				slog.With("useStartTLS", conf.LdapConfig.StartTLS).
+					Debug("Trying to use startTLS")
+				err = c.StartTLS(&tlsConfig)
+				if err != nil {
+					slog.With("err", err).Debug("StartTLS has failed")
+				}
+
+				if !conf.LdapConfig.InsecureNoSSL {
+					slog.With("InsecureNoSSL", conf.LdapConfig.InsecureNoSSL).
+						Debug("Using ldap without tls. The connection will be INSECURE")
+				} else {
+					return nil, errors.New("Can't establish TLS connection and InsecureSkipVerify is false. The connection can't be secure. Aborting")
```

Comment:
Io userei un messaggio conciso tipo solo la prima frase: `Can't establish TLS connection and InsecureSkipVerify is false`. Poi le altre cose vengono aggiunte da dove stampiamo il log se serve direi (tipo l'aborting). Non per forza vogliamo terminare l'esecuzione ecco. Ce in generale quando scriviamo codice ""libreria"" non assumiamo cosa succedera'